admin/app/helpers/solidus_admin/permission_sets_helper.rb

```diff
@@ -0,0 +1,109 @@
+          grouped_permissions[:configuration] << [edit_string, permission.id]
+        when /UserDisplay/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [view_string, permission.id]
+        when /UserManagement/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [edit_string, permission.id]
+        when /RestrictedStockDisplay/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [view_string, permission.id]
+        when /RestrictedStockManagement/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [edit_string, permission.id]
+        when /StockDisplay/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [view_string, permission.id]
+        when /StockManagement/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [edit_string, permission.id]
+        when /ProductDisplay/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [view_string, permission.id]
+        when /ProductManagement/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [edit_string, permission.id]
+        else
+          grouped_permissions[:other] ||= []
+          grouped_permissions[:other] << [permission.name, permission.id]
+        end
+      end
+    end
+
+    # Helper for rendering a set of permission checkboxes.
+    #
+    # @param form [ActionView::Helpers::FormBuilder] The form builder object
+    #   used to generate form elements.
+    # @param title [String] The desired component title.
+    # @param permissions [Array<Array<String, String>>] An array of permission
+    #   pairs, where each pair consists of:
+    #   - A display name for the permission (String).
+    #   - An identifier for the permission (String) which is used as the value
+    #     for the checkbox input.
+    #
+    # @return [String, nil] Returns the generated HTML as a string if
+    #   permissions are present, otherwise returns nil.
+    #
+    # @example
+    #   <%= permission_set_component(
+    #     form: f,
+    #     title: 'Orders',
+    #     permissions: permission_set_options[:order]
+    #   ) %>
+    #   <%= permission_set_component(
+    #     form: f,
+    #     title: 'Products',
+    #     permissions: permission_set_options[:product]
+    #   ) %>
+    def permission_set_component(form:, title:, permissions:)
+      return if permissions.blank?
+
+      content_tag :span, class: "p-2 flex hover:bg-gray-50 rounded-md" do
+        concat(content_tag(:span, title, class: "w-1/2"))
+        concat(
+          content_tag(:span, class: "w-1/2 text-right") do
+            permissions.each do |permission|
+              concat(
+                content_tag(:label, class: "px-2 cursor-pointer") do
```

Comment:
It seems we do not have a checkbox component yet. Could we create one and use it here? It can be done after we merge this PR, but I am a bit worried this could fall over.

Reply:
Done :)